src/NLog/Targets/DatabaseParameterInfo.cs

```diff
@@ -124,6 +129,112 @@ public DatabaseParameterInfo(string parameterName, Layout parameterLayout)
         /// <docgen category='Parameter Options' order='6' />
         [DefaultValue(0)]
         public byte Scale { get; set; }
+
+        internal bool SetDbType(IDbDataParameter dbParameter)
+        {
+            if (!string.IsNullOrEmpty(DbType))
+            {
+                if (_cachedDbTypeSetter == null || !_cachedDbTypeSetter.IsValid(dbParameter.GetType(), DbType))
```

Comment:
maybe nice to reuse `dbParameter.GetType()`

Reply:
It will only make two `GetType()` calls at target-initialization, else the cached _cachedDbTypeSetter  will be valid.